src/Test.UnitTests.Sarif.Multitool.Library/MergeCommandUnitTests.cs

```diff
@@ -0,0 +1,281 @@
+            string outputFilePath5 = Path.Combine(testDirectory, "TESTRULE.005_merged.sarif");
+            StringBuilder outputStringBuilder5 = new StringBuilder();
+            string outputFilePath6 = Path.Combine(testDirectory, "TESTRULE.006_merged.sarif");
+            StringBuilder outputStringBuilder6 = new StringBuilder();
+
+            var mockFileSystem = new Mock<IFileSystem>();
+            ArrangeMockFileSystemRead(mockFileSystem, sarifLog1Json, sarifLog1FilePath);
+            ArrangeMockFileSystemRead(mockFileSystem, sarifLog2Json, sarifLog2FilePath);
+            ArrangeMockFileSystemCreate(mockFileSystem, outputFilePath1, outputStringBuilder1);
+            ArrangeMockFileSystemCreate(mockFileSystem, outputFilePath2, outputStringBuilder2);
+            ArrangeMockFileSystemCreate(mockFileSystem, outputFilePath3, outputStringBuilder3);
+            ArrangeMockFileSystemCreate(mockFileSystem, outputFilePath4, outputStringBuilder4);
+            ArrangeMockFileSystemCreate(mockFileSystem, outputFilePath5, outputStringBuilder5);
+            ArrangeMockFileSystemCreate(mockFileSystem, outputFilePath6, outputStringBuilder6);
+            ArrangeMockFileSystemEnumerate(mockFileSystem, testDirectory, new[] { sarifLog1FilePath, sarifLog2FilePath });
+
+            IFileSystem fileSystem = mockFileSystem.Object;
+
+            var options = new MergeOptions
+            {
+                PrettyPrint = true,
+                OutputDirectoryPath = testDirectory,
+                OutputFileName = "merged.sarif",
+                TargetFileSpecifiers = new[] { "*.sarif" },
+                SplittingStrategy = SplittingStrategy.PerRule,
+                Force = true,
+            };
+
+            var mergeCommand = new MergeCommand(fileSystem);
+            int returnCode = mergeCommand.Run(options);
+            returnCode.Should().Be(0);
+
+            // should have 6 merged logs each log has result of 1 rule
+            List<SarifLog> mergedLogs = new List<SarifLog>
+            {
+                JsonConvert.DeserializeObject<SarifLog>(outputStringBuilder1.ToString()),
+                JsonConvert.DeserializeObject<SarifLog>(outputStringBuilder2.ToString()),
+                JsonConvert.DeserializeObject<SarifLog>(outputStringBuilder3.ToString()),
+                JsonConvert.DeserializeObject<SarifLog>(outputStringBuilder4.ToString()),
+                JsonConvert.DeserializeObject<SarifLog>(outputStringBuilder5.ToString()),
+                JsonConvert.DeserializeObject<SarifLog>(outputStringBuilder6.ToString()),
+            };
+
+            for (int i = 1; i <= 6; i++)
+            {
+                mergedLogs[i - 1].Runs[0].Tool.Driver.Name.Should().Be("TestTool");
+                mergedLogs[i - 1].Runs[0].Results[0].RuleId.Should().Be($"TESTRULE/00{i}");
+            }
+        }
+
+        private static Run CreateTestRun(int numberOfResult, string toolName = null, bool createSubRule = false)
```

Comment:
>CreateTestRun [](http://example.com/codeflow?start=27&length=13)

Don't we have other code somewhere that autogenerates test files? If so, can we please us it instead of creating this duplicative helper?

Reply:
Moved the RandomSarifLogGenerator from Test.UnitTests.Sarif to Test.Utilities.Sarif so can re-use its functionalities.